transport/transport.go

```diff
@@ -506,3 +507,50 @@ func wait(ctx context.Context, closing <-chan struct{}, proceed <-chan int) (int
 		return i, nil
 	}
 }
+
+const (
+	spaceByte   = byte(int(' '))
+	tildaByte   = byte(int('~'))
+	percentByte = byte(int('%'))
+)
+
+// matching https://github.com/grpc/grpc-java/pull/1517/files
+func grpcMessageEncode(grpcMessage string) string {
+	if grpcMessage == "" {
+		return ""
+	}
+	var buffer bytes.Buffer
+	for _, c := range []byte(grpcMessage) {
+		if c >= spaceByte && c < tildaByte && c != percentByte {
+			_ = buffer.WriteByte(c)
+		} else {
+			_, _ = buffer.WriteString(fmt.Sprintf("%%%02X", c))
+		}
+	}
+	return buffer.String()
+}
+
+// matching https://github.com/grpc/grpc-java/pull/1517/files
+func grpcMessageDecode(encodedGrpcMessage string) string {
+	if encodedGrpcMessage == "" {
+		return ""
+	}
+	var buffer bytes.Buffer
+	data := []byte(encodedGrpcMessage)
+	lenData := len(data)
+	for i := 0; i < lenData; i++ {
+		c := data[i]
+		if c == percentByte && i+2 < lenData {
+			parsed, err := strconv.ParseInt(string(data[i+1:i+3]), 16, 8)
+			if err != nil {
+				_ = buffer.WriteByte(c)
```

Comment:
I will if it's a universal thing, but see my last comment on this for @menghanl , will just copy/paste it here:

It's for errcheck, and a style that I like - by doing a blank assignment, I'm explicitly saying "I know there is an error returned here, and I'm purposefully ignoring it, this is not a bug". It tells future contributors that this is not an issue that should be checked, that we are extremely confident this error will not happen and that checking it would be not worthwhile and get in the way of cleaner code (ie no returned error from this function). Does that make sense?